app/src/main/java/com/github/se/travelpouch/ui/dashboard/AddActivityScreen.kt

```diff
@@ -0,0 +1,203 @@
+              label = { Text("Location") },
+              placeholder = { Text("Location") },
+              modifier = Modifier.testTag("locationField"))
+
+          Button(
+              enabled =
+                  location.isNotBlank() &&
+                      title.isNotBlank() &&
+                      description.isNotBlank() &&
+                      dateText.isNotBlank(),
+              onClick = {
+                try {
+                  val finalDate = convertStringToDate(dateText, dateFormat, gregorianCalendar)
+
+                  val activity =
+                      Activity(
+                          activityModelView.getNewUid(),
+                          title,
+                          description,
+                          placeholerLocation,
+                          finalDate,
+                          mapOf())
+
+                  activityModelView.addActivity(activity, context)
+                } catch (e: java.text.ParseException) {
+                  Toast.makeText(
+                          context, "Invalid format, date must be DD/MM/YYYY.", Toast.LENGTH_SHORT)
+                      .show()
+                }
+              },
+              modifier = Modifier.testTag("saveButton")) {
+                Text("Save")
+              }
+        }
+  }
+}
+
+fun convertStringToDate(
+    stringDate: String,
+    dateFormat: SimpleDateFormat,
+    gregorianCalendar: GregorianCalendar
+): Timestamp {
+  val finalDateString =
+      stringDate.substring(0, 2) + "/" + stringDate.substring(2, 4) + "/" + stringDate.substring(4)
+
+  val date = dateFormat.parse(finalDateString)
+  val calendar =
+      gregorianCalendar.apply {
+        time = date!!
+        set(Calendar.HOUR_OF_DAY, 0)
+        set(Calendar.MINUTE, 0)
+        set(Calendar.SECOND, 0)
+      }
+
+  return Timestamp(calendar.time)
+}
+
+class DateVisualTransformation : VisualTransformation {
+  override fun filter(text: AnnotatedString): TransformedText {
+    val transfomedText =
+        text
+            .toString()
+            .mapIndexed { index, c ->
+              when (index) {
+                1 -> "$c/"
+                3 -> "$c/"
+                else -> c
+              }
+            }
+            .joinToString("")
+
+    return TransformedText(
+        text = AnnotatedString(text = transfomedText), offsetMapping = DateOffsetMapping())
+  }
+}
+
+// 00 00 0000
+
+class DateOffsetMapping : OffsetMapping {
+  override fun originalToTransformed(offset: Int): Int {
+    return when (offset) {
+      0 -> 0
+      1 -> 1
+      2 -> 3
+      3 -> 4
+      4 -> 6
+      5 -> 7
+      6 -> 8
+      7 -> 9
+      8 -> 10
+      else -> 10
+    }
+  }
+
+  override fun transformedToOriginal(offset: Int): Int {
+    return when (offset) {
+      0 -> 0
+      1 -> 1
+      2 -> 2
+      3 -> 2
+      4 -> 3
+      5 -> 4
+      6 -> 4
+      7 -> 5
+      8 -> 6
+      9 -> 7
+      10 -> 8
+      else -> 8
+    }
+  }
+}
```

Comment:
Nitpick: I think this could be easily put in another file and documented. I think this is containerisable quite easily and helps to encapsulate some complexity of visual transformers. :3